### PR TITLE
Support eventlet monkey patching in SkipRepeatsQueue

### LINKED ISSUE #332: Watchdog doesn't work along with eventlet
Hi!
Observer cannot be instantiated in case  eventlet.monkey_patch() patches Thread.

Here is the code which shows the problem:
```python
# !/usr/bin/env python

from watchdog.observers import Observer
import eventlet
eventlet.monkey_patch()
obs = Observer() # THIS FAILS because of monkey_patch patches threads
```

Here is the exception I get running the above code
```shell
/usr/bin/python2.7 /home/gnuton/GIT/flickrsmartsync/flickrsmartsync/flickrsmartsync/test.py
Traceback (most recent call last):
  File "/home/gnuton/GIT/flickrsmartsync/flickrsmartsync/flickrsmartsync/test.py", line 6, in <module>
    obs = Observer() # THIS FAILS because of monkey_patch patches threads
  File "/usr/local/lib/python2.7/dist-packages/watchdog/observers/inotify.py", line 186, in **init**
    timeout=timeout)
  File "/usr/local/lib/python2.7/dist-packages/watchdog/observers/api.py", line 208, in **init**
    EventDispatcher.**init**(self, timeout)
  File "/usr/local/lib/python2.7/dist-packages/watchdog/observers/api.py", line 164, in **init**
    self._event_queue = EventQueue()
  File "/usr/lib/python2.7/Queue.py", line 28, in __init__
    self._init(maxsize)
  File "/usr/local/lib/python2.7/dist-packages/watchdog/utils/bricks.py", line 87, in _init
    queue.Queue._init(self, maxsize)
TypeError: unbound method _init() must be called with Queue instance as first argument (got EventQueue instance instead)
```

### PR DESCRIPTION
**Pull request by BoboTiG**
- Introduced the `@cpython_only` marker for tests.
- I used that trick for the fix: https://stackoverflow.com/a/6714313/1117028
- Fixes #332.

### CODE REVIEW

**Comment by BoboTiG** [comment]
Windows tests never starts. Skipping them until #581.

**Comment by BoboTiG** [comment]
@rrzaripov could you review :pray:?

File: tests/test_skip_repeats_queue.py [hunk 1]
```diff
@@ -81,3 +88,14 @@ def test_consecutives_allowed_across_empties():
     q.put(e1)  # this repeat is allowed because 'last' added is now gone from queue
     assert e1 == q.get()
     assert q.empty()
+
+
+@cpython_only
+def test_eventlet_monkey_patching():
```
Comment (BoboTiG) [inline tests/test_skip_repeats_queue.py]: TODO for me: test all functions using the raw and money patched objects with fixtures. Just this test is not enough.

Comment (rrzaripov) [reply tests/test_skip_repeats_queue.py]: Looks good for me, so now I will check how tests run on Windows and patch it if need.

**Pull request merged**